lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -1181,6 +1185,10 @@ bool swift::diagnoseNonSendableTypesInReference(
       if (funcCheckOptions.contains(FunctionCheckKind::Results)) {
         // only check results if funcCheckKind specifies so
         Type resultType = func->getResultInterfaceType().subst(subs);
+
+        if (func->hasSendingResult() && !resultType->hasError())
```

Comment:
I don't think we should bail out early if result is sending, there are a lot of other checks further down in this function that should still pass. Also the return value can't be `true` here, since `true` means that an error was emitted, in our case though there is no error.

Reply:
Thank you.

> Also the return value can't be true here,
It was my misunderstanding🤦🏻‍♂️ Let me fix it.

> We probably shouldn't bail out early if the result is markin with sending, there are a lot of other checks further down in this function that should still pass.

How about checking diag.ID like the below?

```
if (diag.ID == diag::non_sendable_result_in_witness.ID &&
            func->hasSendingResult() && !resultType->hasError())
    return false;
```

non_sendable_result_in_witness.ID is “non-sendable type %0 cannot be returned from %2 implementation to caller of protocol requirement %1”.

Reply:
@azarovalex 

Hi, what do you think about https://github.com/swiftlang/swift/pull/78601#discussion_r1919488417 ?